src/main/kotlin/org/jetbrains/research/testspark/helpers/psi/kotlin/KotlinPsiClassWrapper.kt

```diff
@@ -0,0 +1,125 @@
+            val superClassDescriptor = superClassTypeReference?.let {
+                val bindingContext = it.analyze()
+                bindingContext[BindingContext.TYPE, it]
+            }
+            val superClassPsiClass = superClassDescriptor?.constructor?.declarationDescriptor?.let { descriptor ->
+                DescriptorToSourceUtils.getSourceFromDescriptor(descriptor) as? KtClass
+            }
+            return superClassPsiClass?.let { KotlinPsiClassWrapper(it) }
+        }
+
+    override val virtualFile: VirtualFile get() = psiClass.containingFile.virtualFile
+
+    override val containingFile: PsiFile get() = psiClass.containingFile
+
+    override val fullText: String
+        get() {
+            var fullText = ""
+            val fileText = psiClass.containingFile.text
+
+            // get package
+            packagePattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n\n"
+            }
+
+            // get imports
+            importPattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n"
+            }
+
+            // Add class code
+            fullText += psiClass.text
+
+            return fullText
+        }
+
+    override val classType: ClassType
+        get() {
+            return when {
+                psiClass is KtObjectDeclaration -> ClassType.OBJECT
+                psiClass.isInterfaceClass() -> ClassType.INTERFACE
+                psiClass.hasModifier(KtTokens.ABSTRACT_KEYWORD) -> ClassType.ABSTRACT_CLASS
+                psiClass.isData() -> ClassType.DATA_CLASS
+                psiClass.annotationEntries.any { it.text == "@JvmInline" } -> ClassType.INLINE_VALUE_CLASS
+                else -> ClassType.CLASS
+            }
+        }
+
+    override fun searchSubclasses(project: Project): Collection<PsiClassWrapper> {
+        val scope = GlobalSearchScope.projectScope(project)
+        val lightClass = psiClass.toLightClass()
+        return if (lightClass != null) {
+            val query = ClassInheritorsSearch.search(lightClass, scope, false)
+            query.findAll().map { KotlinPsiClassWrapper(it as KtClass) }
+        } else {
+            emptyList()
+        }
+    }
+
+    override fun getInterestingPsiClassesWithQualifiedNames(
+        psiMethod: PsiMethodWrapper,
+    ): MutableSet<PsiClassWrapper> {
+        val interestingPsiClasses = mutableSetOf<PsiClassWrapper>()
+        val method = psiMethod as KotlinPsiMethodWrapper
+
+        method.psiFunction.valueParameters.forEach { parameter ->
+            val typeReference = parameter.typeReference
+            if (typeReference != null) {
+                val psiClass = PsiTreeUtil.getParentOfType(typeReference, KtClass::class.java)
+                if (psiClass != null && !psiClass.fqName.toString().startsWith("kotlin.")) {
```

Comment:
Here we can check for the nullness of the `qualifiedName` prop and make the `KotlinPsiClassWrapper` throw on PSI classes that do not have `qualifiedName`.

**[relevant to the comment about `qualifiedName` above]**